File: assets/sportic_frames/SplashScreen.java
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import 'package:image_sequence_animator/image_sequence_animator.dart';
import 'package:sportic/Authentication/phn_validation.dart';
import 'package:sportic/Main%20Screens/home.dart';
import 'package:sportic/Providers/secure_storage_service.dart';
import 'package:sportic/main.dart'; // for FCM handler
import 'Providers/splashscreen_provider.dart';
import 'package:firebase_auth/firebase_auth.dart';

class SplashScreen extends ConsumerStatefulWidget {
  const SplashScreen({super.key});

  @override
  ConsumerState<SplashScreen> createState() => _SplashScreenState();
}

class _SplashScreenState extends ConsumerState<SplashScreen> {
  bool _hasNavigated = false;

  Future<void> _handleNavigation(BuildContext context) async {
    if (_hasNavigated) return;
    _hasNavigated = true;

    final user = FirebaseAuth.instance.currentUser;
    final isPhoneLoggedIn = await SecureStorageService.isPhoneLoggedIn();
    final isLoggedIn = user != null || isPhoneLoggedIn;

    final destination = isLoggedIn ? const HomeScreen() : const PhoneLoginScreen();

    setupFCMNotificationListeners(context);

    Navigator.of(context).pushReplacement(
      PageRouteBuilder(
        transitionDuration: const Duration(milliseconds: 1500),
        pageBuilder: (_, __, ___) => destination,
        transitionsBuilder: (_, animation, __, child) {
          return SlideTransition(
            position: Tween<Offset>(
              begin: const Offset(1.0, 0.0),
              end: Offset.zero,
            ).animate(animation),
            child: child,
          );
        },
      ),
    );
  }

  @override
  Widget build(BuildContext context) {
    final isTimerFinished = ref.watch(splashProvider);

    return Scaffold(
      backgroundColor: Colors.black,
      body: Center(
        child: ImageSequenceAnimator(
          'assets/sportic_frames/', // Folder path
          'frame_',                 // Frame name prefix
          0,                        // Start index
          33,                       // End index (for 34 images: 0 to 33)
          'png',                    // File extension
          30,                       // Frame rate (fps)
          isLooping: false,
          isAutoPlay: true,
          onFinish: () {
            if (isTimerFinished) {
              _handleNavigation(context);
            }
          },
          width: MediaQuery.of(context).size.width,
          height: MediaQuery.of(context).size.height,
          fps: 30,
        ),
      ),
    );
  }
}

File: lib/splash_screen.dart
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import 'package:sports/Authentication/phn_validation.dart';

class SplashScreen extends ConsumerStatefulWidget {
  const SplashScreen({super.key});

  @override
  ConsumerState<SplashScreen> createState() => _SplashScreenState();
}

class _SplashScreenState extends ConsumerState<SplashScreen> {
  @override
  void initState() {
    super.initState();

    // Simulate GIF duration (adjust duration as per GIF length, here it's set to 3 seconds)
    Future.delayed(const Duration(milliseconds: 3450), () {
      Navigator.of(context).pushReplacement(
        PageRouteBuilder(
          transitionDuration: const Duration(milliseconds: 100),
          pageBuilder:
              (context, animation, secondaryAnimation) =>
                  const PhoneLoginScreen(),
          transitionsBuilder: (context, animation, secondaryAnimation, child) {
            return SlideTransition(
              position: Tween<Offset>(
                begin: const Offset(1.0, 0.0),
                end: Offset.zero,
              ).animate(animation),
              child: child,
            );
          },
        ),
      );
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.black,
      body: Container(
        width: double.infinity,
        height: double.infinity,
        // decoration: const BoxDecoration(
        //   gradient: LinearGradient(
        //     begin: Alignment.topLeft,
        //     end: Alignment.bottomRight,
        //     colors: [Color(0xFF43325C), Colors.black, Color(0xFF43325C)],
        //   ),
        // ),
        child: Center(
          child: Image.asset(
            'assets/SPORTIC_grad.gif',
            width: 400,
            height: 500,
            fit: BoxFit.cover,
            gaplessPlayback: true,
          ),
        ),
      ),
    );
  }
}
